test/unittests/test_loader.cpp

```diff
@@ -328,3 +324,272 @@ TEST_F(loader, load_and_create_abi_mismatch)
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "x");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(ec, EVMC_LOADER_INVALID_OPTION_NAME);
+    EXPECT_STREQ(evmc_last_error_msg(), "vm_with_set_option (path): unknown option 'x'");
+    EXPECT_EQ(destroy_count, create_count);
+}
+
+TEST_F(loader, load_and_configure_empty_values)
+{
+    supported_options["flag"] = {""};  // Empty value expected.
+    supported_options["e"] = {""};     // Empty value expected.
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,flag,e=,flag=,e", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 4);
+    EXPECT_EQ(recorded_options[0].first, "flag");
+    EXPECT_EQ(recorded_options[0].second, "");
+    EXPECT_EQ(recorded_options[1].first, "e");
+    EXPECT_EQ(recorded_options[1].second, "");
+    EXPECT_EQ(recorded_options[2].first, "flag");
+    EXPECT_EQ(recorded_options[2].second, "");
+    EXPECT_EQ(recorded_options[3].first, "e");
+    EXPECT_EQ(recorded_options[3].second, "");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+    EXPECT_EQ(create_count, 1);
+    EXPECT_EQ(destroy_count, 0);
+}
+
+TEST_F(loader, load_and_configure_degenerated_names)
+{
+    supported_options[""] = {"", "xxx"};  // An option with empty name.
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,,,=,,=xxx", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 5);
+    EXPECT_EQ(recorded_options[0].first, "");
+    EXPECT_EQ(recorded_options[0].second, "");
+    EXPECT_EQ(recorded_options[1].first, "");
+    EXPECT_EQ(recorded_options[1].second, "");
+    EXPECT_EQ(recorded_options[2].first, "");
+    EXPECT_EQ(recorded_options[2].second, "");
+    EXPECT_EQ(recorded_options[3].first, "");
+    EXPECT_EQ(recorded_options[3].second, "");
+    EXPECT_EQ(recorded_options[4].first, "");
+    EXPECT_EQ(recorded_options[4].second, "xxx");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+    evmc_destroy(vm);
+    EXPECT_EQ(destroy_count, create_count);
+}
+
+TEST_F(loader, load_and_configure_comma_at_the_end)
+{
+    // The additional comma at the end of the configuration string is ignored.
+
+    supported_options["x"] = {"x"};
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,x=x,", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 1);
+    EXPECT_EQ(recorded_options[0].first, "x");
+    EXPECT_EQ(recorded_options[0].second, "x");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+    evmc_destroy(vm);
+    EXPECT_EQ(destroy_count, create_count);
+}
+
+TEST_F(loader, load_and_configure_vm_without_set_option)
+{
+    supported_options[""] = {""};  // Allows ,, option.
```

Comment:
This comment seems to be misleading, since below this is an error.

Reply:
Fixed.